runtime/src/zmq-utils.ts

```diff
@@ -281,37 +281,35 @@ export function handleIOPub_status({ msgs, socket, lang }) {
   );
 }
 
-export function handleIOPub_display_data({ msgs, socket }) {
+export function handleIOPub_display_data({ msgs, socket, exec_count }) {
   console.log("emitting display data ..");
   let [podId, name] = msgs.parent_header.msg_id.split("#");
   let payload = {
     podId,
-    name,
+    //name,
     // content is a dict of
     // {
     //   data: {'text/plain': ..., 'image/png': ...},
     //   metadata: {needs_background: 'light'},
     //   transient: ...
     // }
     content: msgs.content,
-    // There's no exe_count in display_data
-    // FIXME I should use execute_reply for count
-    //
-    // count: msgs.content.execution_count,
+    // There's no exec_count in display_data, thus we pass in the session exec_count
+    count: exec_count,
```

Comment:
Could we pass in 0? This value doesn't seem to be useful. Thus, I don't feel maintaining a session exec_count is necessary. It increases the logic complexity.

Reply:
I think the count in `execute_result` is enough.

Reply:
> I think the count in `execute_result` is enough.

The tricky part is that `execute_result` is not always available, instead, IIUC, `execute_reply` would be the last messages in each cell run. The key is how to update the `count` properly, the logic would be either on the frontend or backend.

Reply:
I think we can wait until the final execution results before setting the count. It should be consistent with the behavior of Jupyter?

Reply:
> I think we can wait until the final execution results before setting the count. This should be consistent with the behavior of Jupyter?

Yes, I think a logic in the frontend should work as well.

Reply:
I'm not sure if maintaining `session_exec_count` ourselves is accurate. I'd vote for using `msgs.content.execution_count` because it is always accurate and seems enough for the purpose of showing a count.